### ISSUE TITLE
Support embedded terminals

### ISSUE THREAD

**Issue by Ravenclaw-OIer** [cpeditor/cpeditor#982]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is this feature provided in the master branch?

- [X] I have checked the changelog

### Describe the feature you'd like

This is a somewhat enhancement of #919, instead of adding menu items to allow users to run commands, maybe an embedded terminal (like VSCode's Terminal menu) will help.

Compared to #919 , this gives the user more freedom, and may help if the user has a lot of different commands they *might* run. Though I believe it may cause cross-platform issues, maybe we can add support for GNU/Linux first?

### Are you willing to contribute to this?

- [ ] I'm ready to contribute to this (and will open a PR soon)
- [ ] I'd like to have a try (with the help of the maintainers)
- [ ] No, thanks

**Comment by ouuan** [comment]
> Compared to #919 , this gives the user more freedom

No. Users can start a terminal and cd to the desired directory with #919, so #919 is more powerful, unless you think "embedded" is important and an external terminal is not satisfying.

**Comment by Ravenclaw-OIer** [comment]
> > Compared to #919 , this gives the user more freedom
> 
> No. Users can start a terminal and cd to the desired directory with #919, so #919 is more powerful, unless you think "embedded" is important and an external terminal is not satisfying.

Imagine the following scenario: you have cpeditor on full screen, and you decided to run a terminal via #919. Your terminal pops up in a new window, you finish your commands then click on the text editor, your terminal loses focus immediately and you have to click the cpeditor icon on your (whatever dock) to bring up the interface to select the active window of the app.

Now if it's like that I'd've just clicked the konsole icon.

The point with embedded is that it can be more deeply integrated to the editor, with features like Meta+T to switch to the terminal, etc, etc. 

That's actually why VSCode has a embedded terminal, from my perspective.

**Comment by ouuan** [comment]
> The point with embedded is that it can be more deeply integrated to the editor, with features like Meta+T to switch to the terminal, etc, etc.

That should be the task of a window manager / desktop environment. I think most window managers can switch between two windows via a shortcut.

**Comment by Ravenclaw-OIer** [comment]
Configuring WM is a option, you are right. But my whole point is that *I am using the editor full screen*.

I'll use a screenshot to demonstrate what I am looking for.

![image](https://user-images.githubusercontent.com/44526586/131428671-670f0358-2fda-4456-8d1f-91fb731db380.png)

You can see that in this interface, the code editor and the terminal are displayed simultaneously, which provides a better experience if you need to check your code and execute commands at the same time. True, it is possible that I can align my windows to make them look like this, but imagine I don't want the terminal anymore (for example, I want to see the Messages section of cpeditor), I can switch between them quickly, no window reordering. The terminal is *part of* the editor, it's place can be taken by other parts of the editor, all integrated smoothly.

**Comment by Ravenclaw-OIer** [comment]
As a analogy: I can place several konsole windows together and move them around easily with shortcuts, but I would prefer `tmux` because it is better integrated.
